### PR TITLE
set currentCategoryID

### PR DESCRIPTION
**Pull request by amirjfr**
clicking a category sets the currentCategoryID, dynamically calculating rating and number of raters for each place, getting correct places for each category based on current group id and current category id

### CODE REVIEW

File: client/src/components/CategoryView/CategoryView.js [hunk 1]
```diff
@@ -4,10 +4,20 @@ import { Button, Col, Divider, Row, Typography } from "antd";
...
-  let categoryType = "Restaurants 🍝";
+  const categories = useSelector((state) => state.categories.allCategories);
+  const currentCategoryID = useSelector(
+    (state) => state.categories.currentCategoryID
+  );
+  let currentCategory = categories.find(
+    (category) => category.category_id === currentCategoryID
+  );
+  let categoryType = currentCategory.name + " " + currentCategory.emoji;
+  let btnCategory = currentCategoryID === 1 ? "Beach" : "Restaurant";
```
Comment (sassansh) [inline client/src/components/CategoryView/CategoryView.js:19]: This looks very hard coded. We will be adding other categories later.

File: client/src/components/CategoryView/CategoryView.js [hunk 2]
```diff
@@ -16,7 +26,7 @@ function CategoryView() {
         </Col>
         <Col lg={12} className="addPlaceButton">
           <Button type="primary" icon={<PlusOutlined />} size="large">
-            Add Restaurant
+            Add {btnCategory}
```
Comment (sassansh) [inline client/src/components/CategoryView/CategoryView.js:29]: cant we use `{currentCategory.name}` here?

File: client/src/components/PlaceList/PlaceList.js [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
+      place.category_id === currentCategoryID &&
+      place.group_id === currentGroupID
+  );
 
-  let places = placesData.map((placeData) => (
-    <Place key={placeData.rank} placeData={placeData} />
-  ));
+  let placeList = currentPlaces.map(
+    (placeData) => (
+      <Place key={placeData.place_id} placeData={placeData} rank = {currentPlaces.indexOf(placeData) + 1} />
```
Comment (sassansh) [inline client/src/components/PlaceList/PlaceList.js:20]: Is this rank being properly calculated based on average rating? We want to look at each place, average their review scores then rank them.

Example:
In group 1:
In beaches:

Kitsilano(Ratings: 5,4,3) =4
Ambleside (Ratings: 5,4) =4.5


1.Ambleside
2. Kitsilano

**Review by sassansh — CHANGES_REQUESTED** [review]

File: client/src/components/CategoryView/CategoryView.js (hunk 2 above)
Comment (amirjfr) [reply client/src/components/CategoryView/CategoryView.js:29]: @sassansh the category name is plural that's why I couldn't use it😖

File: client/src/components/PlaceList/PlaceList.js (hunk 3 above)
Comment (amirjfr) [reply client/src/components/PlaceList/PlaceList.js:20]: @sassansh this is just a placeholder until we have more data. I'll change it later.

File: client/src/components/CategoryView/CategoryView.js (hunk 2 above)
Comment (sassansh) [reply client/src/components/CategoryView/CategoryView.js:29]: ahh fine! Good for now then. Well think of a solution later

File: client/src/components/PlaceList/PlaceList.js (hunk 3 above)
Comment (sassansh) [reply client/src/components/PlaceList/PlaceList.js:20]: kk, as long as were aware! go ahead and merge then :) great job!

**Review by sassansh — APPROVED** [review]

File: client/src/components/CategoryView/CategoryView.js (hunk 1 above)
Comment (amirjfr) [reply client/src/components/CategoryView/CategoryView.js:19]: @sassansh yes we have to think of something later.

File: client/src/components/PlaceList/PlaceList.js (hunk 3 above)
Comment (amirjfr) [reply client/src/components/PlaceList/PlaceList.js:20]: @sassansh Thanks😃

Comment (johnnybcs) [reply client/src/components/PlaceList/PlaceList.js:20]: Hey guys just thought I'd drop by and say hi. Getting so many notifications, didn't want to miss out on the action.

Comment (amirjfr) [reply client/src/components/PlaceList/PlaceList.js:20]: @johnnybcs lol😂

**Pull request merged**